src/main/java/teammates/ui/webapi/output/CommentSearchResultData.java

```diff
@@ -0,0 +1,75 @@
+package teammates.ui.webapi.output;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import teammates.common.datatransfer.FeedbackResponseCommentSearchResultBundle;
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+
+/**
+ * API output format for a comment search result.
+ */
+public class CommentSearchResultData extends SessionResultsData {
+    private final FeedbackSessionData feedbackSession;
+
+    public CommentSearchResultData(FeedbackSessionAttributes session,
+                                   FeedbackResponseCommentSearchResultBundle bundle) {
+        super();
+        feedbackSession = new FeedbackSessionData(session);
+        bundle.questions.forEach((key, value) -> {
+            questions.addAll(buildQuestions(value, bundle));
+        });
+    }
+
+    private List<QuestionOutput> buildQuestions(List<FeedbackQuestionAttributes> questions,
+                                                FeedbackResponseCommentSearchResultBundle bundle) {
+        List<QuestionOutput> output = new ArrayList<>();
+        for (FeedbackQuestionAttributes question : questions) {
+            output.add(new QuestionOutput(question, null,
+                    buildResponses(bundle.responses.get(question.getId()), bundle)));
+        }
+        return output;
+    }
+
+    private List<ResponseOutput> buildResponses(List<FeedbackResponseAttributes> responses,
+                                                FeedbackResponseCommentSearchResultBundle bundle) {
+        List<ResponseOutput> output = new ArrayList<>();
+        for (FeedbackResponseAttributes response : responses) {
+            output.add(new ResponseOutput(response.getId(), response.giver, null, null, null,
```

Comment:
There is a nice builder for `ResponseOutput` merged recently. You can take advantage on that.